### PR TITLE
tidb-lightning: document backend and that system DBs are filtered

### PR DESCRIPTION
**Pull request by anotherrachel**
### What is changed, added or deleted? 
This PR
* adds document changes introduced in TiDB Lightning v3.0.5.
* ~updates the download links of Lightning to the latest version.~ (updated in https://github.com/pingcap/docs-cn/pull/2165)
* adds a doc for TiDB Lightning TiDB-Backend.
* updates config to include changes from https://github.com/pingcap/tidb-lightning/pull/248
* updates the required privileges in FAQ to include TiDB-backend.

### What is the related PR or file link(s)? 
https://github.com/pingcap/docs/pull/1620/
https://github.com/pingcap/docs/pull/1693/ (update glossary in https://github.com/pingcap/docs-cn/pull/2214)

### Which version does your change affect? 
dev, v3.0 and v3.1

### CODE REVIEW

**Comment by anotherrachel** [comment]
@kennytm @yikeke PTAL, thanks!

File: dev/faq/tidb-lightning.md [hunk 1]
```diff
@@ -23,11 +23,13 @@ TiDB Lightning 需要以下权限：
 * CREATE
 * DROP
 
-存储断点的数据库额外需要以下权限：
+如果选择 [TiDB-backend](/dev/reference/tools/tidb-lightning/tidb-backend.md) 模式，或目标数据库用于存储断点，则 TiBD Lightning 额外需要以下权限：
 
 * INSERT
 * DELETE
 
++Importer-backend 无需以上两个权限，因为数据直接被 Ingest 到 TiKV 中，所以绕过了 TiBD 的权限系统。只要 TiKV、TiKV Importer 和 TiDB Lightning 的端口在集群之外不可访问，就可以保证安全。
```
Comment (yikeke) [inline dev/faq/tidb-lightning.md]:
```suggestion
+Importer-backend 无需以上两个权限，因为数据直接被 Ingest 到 TiKV 中，所以绕过了 TiDB 的权限系统。只要 TiKV、TiKV Importer 和 TiDB Lightning 的端口在集群之外不可访问，就可以保证安全。
```

File: dev/reference/tools/tidb-lightning/config.md [hunk 2]
```diff
@@ -84,8 +84,15 @@ driver = "file"
 # keep-after-success = false
 
 [tikv-importer]
-# tikv-importer 的监听地址，需改为实际地址。
+# 选择后端：“importer” 或 “tidb“
+# backend = "importer"
+# 当后端是 “importer” 时，tikv-importer 的监听地址（需改为实际地址）。
 addr = "172.16.31.10:8287"
+# 当后端是 “tidb” 时，插入重复数据时执行的操作。
+# - replace：新数据题替代已有数据
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/config.md]:
```suggestion
# - replace：新数据替代已有数据
```

File: dev/reference/tools/tidb-lightning/deployment.md [hunk 3]
```diff
@@ -5,7 +5,9 @@ category: reference
 
 # TiDB Lightning 部署与执行
 
-本文主要介绍 TiDB Lightning 单独部署与混合部署的硬件需求，以及使用 Ansible 部署与手动部署这两种部署方式。
+本文主要介绍 TiDB Lightning 使用 Importer-backend （默认）进行数据导入的硬件需求，以及使用 Ansible 部署与手动部署 TiDB Lightning 这两种部署方式。
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/deployment.md]:
```suggestion
本文主要介绍 TiDB Lightning 使用 Importer-backend（默认）进行数据导入的硬件需求，以及使用 Ansible 部署与手动部署 TiDB Lightning 这两种部署方式。
```

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
+---
+title: TiDB Lightning TiDB-Backend
+summary: 了解 TiDB Lightning TiDB-backend。
+category: reference
+---
+
+# TiDB Lightning TiDB-Backend
+
+后端决定 `tidb-lightning` 将如何把将数据导入到目标集群中。目前，TiDB Lightning 支持 Importer-backend （默认）和 TiDB-backend 两种后端，两者导入数据的区别如下：
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
TiDB Lightning 的后端决定 `tidb-lightning` 将如何把将数据导入到目标集群中。目前，TiDB Lightning 支持 Importer-backend（默认）和 TiDB-backend 两种后端，两者导入数据的区别如下：
```

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+* **Importer-backend**：`tidb-lightning` 先将 SQL 或 CSV 数据编码成键值对，由 `tikv-importer` 对写入的键值对进行排序，然后把这些键值对 Ingest 到 TiKV 节点中。
+
+* **TiDB-backend**：`tidb-lightning` 先将数据编码到 SQL 的 `INSERT` 语句中，然后直接在 TiDB 节点上运行这些语句进行数据导入。
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
* **TiDB-backend**：`tidb-lightning` 先将数据编码成 `INSERT` 语句，然后直接在 TiDB 节点上运行这些 SQL 语句进行数据导入。
```

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+## 部署 TiDB-backend
+
+使用 TiDB-backend 时，你无需部署 `tikv-importer`。与[标准部署过程](/dev/reference/tools/tidb-lightning/deployment.md)相比，部署 TiDB-backend 时有如下不同：
+
+* 可以跳过所有涉及 `tikv-importer` 的步骤。
+* 必须更改相应配置申明使用的是 TiDB-backend。
+
+### 硬件需求
+
+使用 TiDB-backend 时， TiDB Lightning 的速度仅受限于 TiDB 执行 SQL 语句时的速度。因此，即使是低配的机器也可能发挥出最佳性能。推荐的硬件配置如下：
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
使用 TiDB-backend 时， TiDB Lightning 的速度仅受限于 TiDB 执行 SQL 语句的速度。因此，即使是低配的机器也可能发挥出最佳性能。推荐的硬件配置如下：
```

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    [importer_server]
+    # keep empty
+
+    [lightning_server]
+    192.168.20.10
+
+    ...
+    ```
+
+2. 忽略 `group_vars/all.yml` 文件中 `tikv_importer_port` 部分的设置，`group_vars/importer_server.yml` 文件也不需要修改。但是你需要在 `conf/tidb-lightning.yml` 文件中将 `backend` 设置更改为`tidb`。
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
2. 忽略 `group_vars/all.yml` 文件中 `tikv_importer_port` 部分的设置，`group_vars/importer_server.yml` 文件也不需要修改。但是你需要在 `conf/tidb-lightning.yml` 文件中将 `backend` 设置更改为 `tidb`。
```

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+
+3. 启动、部署集群。
+
+4. 为 TiDB Lightning 挂载数据源。
+
+5. 启动 `tidb-lightning`。
+
+### 手动部署
+
+手动部署时，你无需下载 `tikv-importer` 的配置文件进行配置。
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
手动部署时，你无需下载和配置 `tikv-importer`。
```
原文意思是哪种？这个向作者确认后才改的吗？

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+```toml
+[tikv-importer]
+backend = "tidb"
+```
+
+或者在用命令行启动 `tidb-lightning` 时，传入参数 `--backend tidb`。
+
+## 冲突解决
+
+TiDB-backend 支持导入已填充的表（非空表）。但是，新数据可能会与旧数据的唯一键冲突。你可以通过使用如下任务配置来调整遇到冲突时的默认行为：
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
TiDB-backend 支持导入到已填充的表（非空表）。但是，新数据可能会与旧数据的唯一键冲突。你可以通过使用如下任务配置来控制遇到冲突时的默认行为：
```

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+
+| 设置 | 冲突时默认行为 | 对应 SQL 语句 |
+|:---|:---|:---|
+| replace | 新数据替代旧数据 | `REPLACE INTO ...` |
+| ignore | 保留旧数据，忽略新数据 | `INSERT IGNORE INTO ...` |
+| error | 中止导入 | `INSERT INTO ...` |
+
+## 从 Loader 迁移到 TiDB Lightning TiDB-backend
+
+TiDB Lightning TiDB-backend 可以替代 [Loader](/dev/reference/tools/loader.md)。下表说明了如何将 [Loader](/dev/reference/tools/loader.md) 的配置迁移到 [TiDB Lightning 配置](/dev/reference/tools/tidb-lightning/config.md)中：
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
TiDB Lightning TiDB-backend 可以完全取代 [Loader](/dev/reference/tools/loader.md)。下表说明了如何将 [Loader](/dev/reference/tools/loader.md) 的配置迁移到 [TiDB Lightning 配置](/dev/reference/tools/tidb-lightning/config.md)中：
```

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+</td><td>
+
+```toml
+[lightning]
+# 日志
+level = "info"
+file = "tidb-lightning.log"
+# Prometheus
+pprof-port = 8289
+# 线程数 (最好使用默认设置)
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
# 并发度 (最好使用默认设置)
```
这里也是向作者确认过的吗

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+
+</td><td>
+
+```toml
+[checkpoint]
+# 断点存储
+enable = true
+schema = "tidb_lightning_checkpoint"
+# 断点默认存储在本地的文件系统，这样更高效。但你也可以
+# 选择将断点存储在兼容 MySQL 但数据库服务器中，设置如下：
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
# 选择将断点存储在目标数据库中，设置如下：
```

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+```
+
+</td><td>
+
+```toml
+[tidb]
+# TiDB 连接参数
+host = "127.0.0.1"
+port = 4000
+status-port = 10080  # <- 必须
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
status-port = 10080  # <- 必须有的参数
```

File: dev/TOC.md [hunk 14]
```diff
@@ -344,6 +344,7 @@
       - [断点续传](/dev/reference/tools/tidb-lightning/checkpoints.md)
       - [表库过滤](/dev/reference/tools/tidb-lightning/table-filter.md)
       - [CSV 支持](/dev/reference/tools/tidb-lightning/csv.md)
+      - [TiDB-backend](/dev/reference/tools/tidb-lightning/tidb-backend.md)
```
Comment (kennytm) [inline dev/TOC.md]:
```suggestion
      - [TiDB 后端](/dev/reference/tools/tidb-lightning/tidb-backend.md)
```

Either use "后端" for all "backend"...

File: dev/reference/tools/tidb-lightning/config.md [hunk 15]
```diff
@@ -84,8 +84,15 @@ driver = "file"
 # keep-after-success = false
 
 [tikv-importer]
-# tikv-importer 的监听地址，需改为实际地址。
+# 选择后端：“importer” 或 “tidb“
+# backend = "importer"
+# 当后端是 “importer” 时，tikv-importer 的监听地址（需改为实际地址）。
 addr = "172.16.31.10:8287"
+# 当后端是 “tidb” 时，插入重复数据时执行的操作。
```
Comment (kennytm) [inline dev/reference/tools/tidb-lightning/config.md]: ... or don't translate the "backend" here.

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 16]
```diff
@@ -0,0 +1,208 @@
...
+## 部署 TiDB-backend
+
+使用 TiDB-backend 时，你无需部署 `tikv-importer`。与[标准部署过程](/dev/reference/tools/tidb-lightning/deployment.md)相比，部署 TiDB-backend 时有如下不同：
+
+* 可以跳过所有涉及 `tikv-importer` 的步骤。
+* 必须更改相应配置申明使用的是 TiDB-backend。
+
+### 硬件需求
+
+使用 TiDB-backend 时， TiDB Lightning 的速度仅受限于 TiDB 执行 SQL 语句的速度。因此，即使是低配的机器也可能发挥出最佳性能。推荐的硬件配置如下：
```
Comment (kennytm) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
使用 TiDB-backend 时， TiDB Lightning 的速度仅受限于 TiDB 执行 SQL 语句的速度。因此，即使是低配的机器也足够发挥出最佳性能。推荐的硬件配置如下：
```

File: dev/TOC.md (hunk 14 above)
Comment (anotherrachel) [reply dev/TOC.md]: @kennytm 我的想法是泛指 backend 的时候，中文直译成“后端”，特指的时候，就使用“TiDB-backend”，不知是否可行？ @yikeke 有什么建议嘛？

File: dev/reference/tools/tidb-lightning/tidb-backend.md [hunk 17]
```diff
@@ -0,0 +1,208 @@
...
+
+# TiDB Lightning TiDB-Backend
+
+TiDB Lightning 的后端决定 `tidb-lightning` 将如何把将数据导入到目标集群中。目前，TiDB Lightning 支持 Importer-backend（默认）和 TiDB-backend 两种后端，两者导入数据的区别如下：
+
+* **Importer-backend**：`tidb-lightning` 先将 SQL 或 CSV 数据编码成键值对，由 `tikv-importer` 对写入的键值对进行排序，然后把这些键值对 Ingest 到 TiKV 节点中。
+
+* **TiDB-backend**：`tidb-lightning` 先将数据编码成 `INSERT` 语句，然后直接在 TiDB 节点上运行这些 SQL 语句进行数据导入。
+
+| 后端 | Importer | TiDB |
```
Comment (yikeke) [inline dev/reference/tools/tidb-lightning/tidb-backend.md]:
```suggestion
| 后端 | Importer-backend | TiDB-backend |
```

File: dev/TOC.md (hunk 14 above)
Comment (yikeke) [reply dev/TOC.md]: 嗯我也建议 backend 单独出现的时候译出来，如果出现的是 TiDB-backend 或 Importer-backend 统一不译～

**Comment by anotherrachel** [comment]
@yikeke @kennytm v3.0 and v3.1 updated. PTAL, thanks!

**Comment by WangXiangUSTC** [comment]
why ignore v2.1?

**Comment by kennytm** [comment]
Most of these features do not exist in 2.1.

**Pull request merged**